Assets/MixedRealityToolkit.SDK/Experimental/Dialog/Scripts/Dialog.cs

```diff
@@ -0,0 +1,232 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Experimental.Dialog
+{
+    public abstract class Dialog : MonoBehaviour
+    {
+        protected DialogResult result;
+
+        protected DialogState state = DialogState.Uninitialized;
+        public DialogState State
+        {
+            get
+            {
+                return state;
+            }
+            set
+            {
+                state = value;
+            }
+        }
+
+        /// <summary>
+        /// Called after user has clicked a button and the dialog has finished closing
+        /// </summary>
+        public Action<DialogResult> OnClosed;
+
+        /// <summary>
+        /// Can be used to monitor result instead of events
+        /// </summary>
+        public DialogResult Result
+        {
+            get
+            {
+                return result;
+            }
+        }
+
+        protected void Launch(DialogResult newResult)
+        {
+            if (State != DialogState.Uninitialized)
+            {
+                return;
+            }
+
+            result = newResult;
+            StartCoroutine(RunDialogOverTime());
+        }
+
+        /// <summary>
+        /// Opens dialog, waits for input, then closes
+        /// </summary>
+        /// <returns></returns>
```

Comment:
Remove returns tag or fill it in. Empty returns will fail automatic PR pipeline validation